cloudant-client/src/test/java/com/cloudant/tests/CloudFoundryServiceTest.java

```diff
@@ -0,0 +1,170 @@
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewService("vcapSingleService",
+                        CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix("vcapSingleService").build().serverVersion();
+    }
+
+    @Test
+    public void vcapSingleServiceNoName() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix().build().serverVersion();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void vcapSingleServiceMissingService() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                return new CloudFoundryServiceTest.VCAPGenerator().toJson();
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void vcapSingleServiceMultipleServiceWithNoNameSpecified() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                VCAPGenerator vcap = new CloudFoundryServiceTest.VCAPGenerator();
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                vcap.createNewServiceWithoutName(CloudantClientHelper.SERVER_URI_WITH_USER_INFO,
+                        CloudantClientHelper.COUCH_USERNAME, CloudantClientHelper.COUCH_PASSWORD);
+                return vcap.toJson();
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void vcapInvalidJSON() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                return "{\"cloudantNoSQLDB\":[]"; // invalid JSON
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void vcapNotPresent() {
+        new MockUp<System>() {
+            @Mock
+            public String getenv(final String string) {
+                return null;
+            }
+        };
+        ClientBuilder.bluemix();
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
As per findbugs, if the VCAP JSON has no `credentials` this could NPE and it appears this test catches that case. I think it should probably be `IllegalStateException` again.